pkg/operator/ceph/cluster/osd/osd_test.go

```diff
@@ -514,3 +528,395 @@ func TestGetOSDInfoWithCustomRoot(t *testing.T) {
+	// 5 deployments should have been created
+	assert.Len(t, deploymentCreateTimes, 5)
+	assert.Len(t, deploymentUpdateTimes, 0)
+	logger.Info("deployments successfully created for new PVCs")
+
+	// =============================================================================================
+	logger.Info("Step 2: verify deployments are updated when run again")
+	// clean the create times maps
+	clearCreateTimes := func() {
+		for k := range deploymentCreateTimes {
+			delete(deploymentCreateTimes, k)
+		}
+		for k := range deploymentUpdateTimes {
+			delete(deploymentUpdateTimes, k)
+		}
+	}
+	clearCreateTimes()
+	statusMapWatcher.Reset() // fake 'watcher' can close the channel for long tests, so reset when we can
+
+	c = New(context, clusterInfo, spec, "myversion")
+	go func() {
+		provisionConfig := c.newProvisionConfig()
+		c.startProvisioningOverPVCs(provisionConfig)
+	}()
+
+	pvcList = waitForNumPVCs(numExpectedPVCs)
+	sort.Strings(pvcList)
+	for i, pvc := range pvcList {
+		mockNodeOrchestrationCompletion(c, pvc, i, statusMapWatcher)
+	}
+	logger.Info("done marking PVCs completed")
+
+	waitForDeploymentUpdates(numExpectedPVCs)
+	assert.Len(t, deploymentUpdateTimes, 5)
+	// updates are an attempt to create (returns AlreadyExists error) followed by an update
+	assert.Len(t, deploymentCreateTimes, 5)
+
+	logger.Info("deployment creates:", deploymentCreateTimes)
+	logger.Info("deployment updates:", deploymentUpdateTimes)
+
+	assertUpdatesAfterCreates := func(expectedCreates int) {
+		firstUpdate := time.Now() // find the when the first deployment update happened
+		for depName, updated := range deploymentUpdateTimes {
+			if updated.Before(firstUpdate) {
+				firstUpdate = updated // find first deployment update
+			}
+			// every update is preceded by a create attempt which returns AlreadyExists error
+			created := deploymentCreateTimes[depName]
+			assert.True(t, updated.After(created))
+			// for later processing, limit the deploymentCreateTimes to only those without updates
+			delete(deploymentCreateTimes, depName)
+		}
+		finalCreate := time.Time{} // find when the final create (without subsequent update) happened
+		for _, created := range deploymentCreateTimes {
+			if created.After(finalCreate) {
+				finalCreate = created
+			}
+		}
+		// ensure that the first update happens after the final create operation
+		assert.True(t, firstUpdate.After(finalCreate))
+		assert.Len(t, deploymentCreateTimes, expectedCreates)
+	}
+
+	assertUpdatesAfterCreates(0)
+
+	// =============================================================================================
+	logger.Info("Step 3: verify new deployments are created before existing ones are updated")
+	clearCreateTimes()
+	spec.Storage.StorageClassDeviceSets[0].Count = 8
+	numExpectedPVCs = 8
+	statusMapWatcher.Reset() // fake 'watcher' can close the channel for long tests, so reset when we can
+
+	c = New(context, clusterInfo, spec, "myversion")
+	go func() {
```

Comment:
Is the go routine exiting at some point? How do we assert there is no error?

Reply:
It does not exit but waits for a test timeout if things fail... testing with VSCode there is an automatic 30 sec timeout, but I should code a timeout for this for our CI. Good catch.